server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/meta/MetricEntry.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.fileformats.meta;
+
+import org.opensearch.index.compositeindex.datacube.MetricStat;
+
+import java.util.Objects;
+
+/**
+ * Holds the pair of metric name and it's associated stat
+ *
+ * @opensearch.experimental
+ */
+public class MetricEntry {
```

Comment:
Can't we use  `org.opensearch.index.compositeindex.datacube.Metric`? What advantages do you see with this flatenned kind of abstraction? Also, there is nothing storage specific read/write here, right?

Reply:
Used Metric now!